src/code/z_actor.c

```diff
@@ -456,232 +469,236 @@ void Target_SetColors(TargetContext* targetCtx, Actor* actor, s32 type, PlayStat
+        }
 
-            Target_SetPos(targetCtx, targetCtx->unk4C, projectedPos.x, projectedPos.y, projectedPos.z);
+        Target_SetLockOnPos(targetCtx, targetCtx->currentLockOnIndex, projectedPos.x, projectedPos.y, projectedPos.z);
 
-            if ((!(player->stateFlags1 & PLAYER_STATE1_40)) || (actor != player->targetedActor)) {
-                OVERLAY_DISP = Gfx_SetupDL(OVERLAY_DISP, SETUPDL_57);
+        if (!(player->stateFlags1 & PLAYER_STATE1_40) || (actor != player->targetedActor)) {
+            OVERLAY_DISP = Gfx_SetupDL(OVERLAY_DISP, SETUPDL_57);
 
-                for (spB0 = 0, spAC = targetCtx->unk4C; spB0 < spB8; spB0++, spAC = (spAC + 1) % 3) {
-                    entry = &targetCtx->unk50[spAC];
+            for (i = 0, index = targetCtx->currentLockOnIndex; i < totalEntries;
+                 i++, index = (index + 1) % ARRAY_COUNT(targetCtx->lockOnEntries)) {
+                entry = &targetCtx->lockOnEntries[index];
 
-                    if (entry->unkC < 500.0f) {
-                        if (entry->unkC <= 120.0f) {
-                            var2 = 0.15f;
-                        } else {
-                            var2 = ((entry->unkC - 120.0f) * 0.001f) + 0.15f;
-                        }
+                if (entry->distance < 500.0f) {
+                    s32 lockOnIndex;
+
+                    if (entry->distance <= 120.0f) {
+                        var2 = 0.15f;
+                    } else {
+                        var2 = ((entry->distance - 120.0f) * 0.001f) + 0.15f;
+                    }
 
-                        Matrix_Translate(entry->pos.x, entry->pos.y, 0.0f, MTXMODE_NEW);
-                        Matrix_Scale(var2, 0.15f, 1.0f, MTXMODE_APPLY);
+                    Matrix_Translate(entry->pos.x, entry->pos.y, 0.0f, MTXMODE_NEW);
+                    Matrix_Scale(var2, 0.15f, 1.0f, MTXMODE_APPLY);
 
-                        gDPSetPrimColor(OVERLAY_DISP++, 0, 0, entry->color.r, entry->color.g, entry->color.b,
-                                        (u8)alpha);
+                    gDPSetPrimColor(OVERLAY_DISP++, 0, 0, entry->color.r, entry->color.g, entry->color.b, (u8)alpha);
 
-                        Matrix_RotateZS((targetCtx->unk4B * 512), MTXMODE_APPLY);
+                    Matrix_RotateZS(targetCtx->rotation * 512, MTXMODE_APPLY);
 
-                        for (i = 0; i < 4; i++) {
-                            Matrix_RotateZS(0x4000, MTXMODE_APPLY);
-                            Matrix_Push();
-                            Matrix_Translate(entry->unkC, entry->unkC, 0.0f, MTXMODE_APPLY);
-                            gSPMatrix(OVERLAY_DISP++, Matrix_NewMtx(play->state.gfxCtx), G_MTX_MODELVIEW | G_MTX_LOAD);
-                            gSPDisplayList(OVERLAY_DISP++, gZTargetLockOnTriangleDL);
-                            Matrix_Pop();
-                        }
+                    // Draw the 4 lock-on triangles
+                    for (lockOnIndex = 0; lockOnIndex < 4; lockOnIndex++) {
```

Comment:
Would `triangleIndex` make more sense? `lockOnIndex` is already used to refer to the entries, not the triangles